server/src/main/java/org/elasticsearch/common/time/Iso8601Parser.java

```diff
@@ -487,10 +494,32 @@ private ZoneId parseZoneId(CharSequence str, int pos) {
         if (seconds == null || seconds > 59) return null;
         if (len == pos) return ofHoursMinutesSeconds(hours, minutes, seconds, positive);
 
+        if (str.charAt(pos) == '[' && str.charAt(len - 1) == ']') {
+            return parseRawZoneId(str, pos);
+        }
+
         // there's some text left over...
         return null;
     }
 
+    /**
+     * Parses a raw zone id, which is a string of the form [zoneId] (eg [Europe/Paris]).
+     *
+     * @param str The string to parse
+     * @param pos The position in the string where the zone id starts (the first character after the opening [)
+     * @return The parsed zone id, or {@code null} if the string is not a valid zone id.
+     */
+    private ZoneId parseRawZoneId(CharSequence str, int pos) {
+        try {
+            String zoneId = str.subSequence(pos + 1, str.length() - 1).toString();
+            // Try to resolve short zone ids to offsets.
+            zoneId = ZoneId.SHORT_IDS.getOrDefault(zoneId, zoneId);
+            return ZoneId.of(zoneId);
+        } catch (DateTimeException e) {
+            return null;
```

Comment:
What about unknown suffixes, or suffixes representing features like e.g. calendars? (for example,  `2022-07-08T00:14:07+01:00[knort=blargel]`?) 
I think that in that case we would want to ignore that, and keep the parsed offset. Here instead I think we end up ignoring both.